src/lib.rs

```diff
@@ -97,7 +97,7 @@ mod histbuf;
 mod indexmap;
 mod indexset;
 mod linear_map;
-mod string;
+pub mod string;
```

Comment:
why make this module public? String is already reexported at the crate root.

Reply:
That's for the new `FromUtf16Error` type.

This does bring up a good point though, `string::format` should have `#[doc(hidden)]` to keep it out-of-sight if this module is public.

Reply:
ah okay. it feels a bit strange to me that you can now refer to `String` via either `heapless::String` or `heapless::string::String` though. Maybe we should remove the reexport from the crate root, so all string stuff is available on `heapless::string::*` only? It's a breaking change though, and we just released v0.8.

(and maybe we should do it for other stuff. the map stuff is cluttering the crate root quite a bit too)

Reply:
I think that's a good idea, but for now it doesn't hurt to have it available from two locations, and it keeps it non-breaking.

Reply:
okay, sounds good to me then.